src/Models/HydrostaticFreeSurfaceModels/hydrostatic_free_surface_model.jl

```diff
@@ -100,30 +101,43 @@ Keyword arguments
   - `velocities`: The model velocities. Default: `nothing`.
   - `pressure`: Hydrostatic pressure field. Default: `nothing`.
   - `diffusivity_fields`: Diffusivity fields. Default: `nothing`.
+  - `vertical_coordinate`: choice between the default `ZCoordinate` and the free-surface following `ZStarSpacing`
   - `auxiliary_fields`: `NamedTuple` of auxiliary fields. Default: `nothing`.
 """
 function HydrostaticFreeSurfaceModel(; grid,
-                                             clock = Clock{eltype(grid)}(time = 0),
-                                momentum_advection = CenteredSecondOrder(),
-                                  tracer_advection = CenteredSecondOrder(),
-                                          buoyancy = nothing,
-                                          coriolis = nothing,
-                                      free_surface = default_free_surface(grid, gravitational_acceleration=g_Earth),
-                                           tracers = nothing,
-                                      forcing::NamedTuple = NamedTuple(),
-                                           closure = nothing,
-                   boundary_conditions::NamedTuple = NamedTuple(),
-                     particles::ParticlesOrNothing = nothing,
-             biogeochemistry::AbstractBGCOrNothing = nothing,
-                                        velocities = nothing,
-                                          pressure = nothing,
-                                diffusivity_fields = nothing,
-                                  auxiliary_fields = NamedTuple()
+                                     clock = Clock{eltype(grid)}(time = 0),
+                        momentum_advection = CenteredSecondOrder(),
+                          tracer_advection = CenteredSecondOrder(),
+                                  buoyancy = SeawaterBuoyancy(eltype(grid)),
+                                  coriolis = nothing,
+                              free_surface = default_free_surface(grid, gravitational_acceleration=g_Earth),
+                                   tracers = (:T, :S),
+                       forcing::NamedTuple = NamedTuple(),
+                                   closure = nothing,
+           boundary_conditions::NamedTuple = NamedTuple(),
+             particles::ParticlesOrNothing = nothing,
+     biogeochemistry::AbstractBGCOrNothing = nothing,
+                                velocities = nothing,
+                                  pressure = nothing,
+                        diffusivity_fields = nothing,
+           generalized_vertical_coordinate = nothing,
```

Comment:
`generalized` is implied fine here, so I would go with the name `vertical_coordinate`.